util/CheckSums.h

```diff
@@ -19,6 +19,14 @@ namespace CheckSums {
     constexpr bool csc_float = noexcept(noexcept(std::log10(std::abs(43.0f))));
     FO_COMMON_API void CheckSumCombine(uint32_t& sum, float t) noexcept(csc_float);
 
+    // unify char checksum for archs with singed and unsigned char
+    constexpr void CheckSumCombine(uint32_t& sum, char c)
+    {
+        signed char t = static_cast<signed char>(c);
```

Comment:
doesn't produce consistent results for `signed char` and `unsigned char` constructed with values outside the range 0-127, eg:
````
    static constexpr auto char_cs2a = []() {
        uint32_t sum = 0;
        constexpr auto num = static_cast<signed char>(195);
        CheckSumCombine(sum, num);
        return sum;
    }();
    static constexpr auto char_cs2b = []() {
        uint32_t sum = 0;
        constexpr auto num = static_cast<unsigned char>(195);
        CheckSumCombine(sum, num);
        return sum;
    }();

    static constexpr auto char_cs3a = []() {
        uint32_t sum = 0;
        constexpr auto num = static_cast<signed char>(-15);
        CheckSumCombine(sum, num);
        return sum;
    }();
    static constexpr auto char_cs3b = []() {
        uint32_t sum = 0;
        constexpr auto num = static_cast<unsigned char>(-15);
        CheckSumCombine(sum, num);
        return sum;
    }();

    static_assert(char_cs2a == 61);
    static_assert(char_cs2b == 61);
    static_assert(char_cs3a == 15);
    static_assert(char_cs3b == 15);
````

Reply:
Maybe it because we change sign but instead should use 2's complement for consistency?

Reply:
Has C++ chars somehow knowledge about UTF8, UTF16 and UTF32(maybe in some Asian langs or special chars like smiles or similar glyphs)?

Reply:
No, there other types for Unicode characters in C++.